assets/model_monitoring/components/src/model_monitor_metric_outputter/runmetric_client.py

```diff
@@ -0,0 +1,157 @@
+                    run_name=run_name,
+                    tags=self._create_run_tags(
+                        monitor_name, None, "azureml.metrics", []
+                    ),
+                )
+                .info.run_id
+            )
+            with mlflow.start_run(run_id=metric_run_id):
+                print(
+                    f"Creating parent metric run with name '{run_name}' and id '{metric_run_id}'."
+                )
+        else:
+            metric_run_id = runs.iloc[0].run_id
+            print(f"Found run with id '{metric_run_id}' matching filter.")
+
+        self.parent_run_id = metric_run_id
+
+        return metric_run_id
+
+    def get_or_create_run_id(
+        self, monitor_name: str, signal_name: str, metric_name: str, groups: List[str]
+    ) -> str:
+        """Get or create a run id for a given monitor, signal, metric and groups."""
+        experiment_id = self._get_experiment_id()
+        if experiment_id is None:
+            print("No experiment id found. Skipping publishing run metrics.")
+            return None
+        filter_query = self._create_filter_query(
+            monitor_name, signal_name, metric_name, groups
+        )
+        print(f"Fetching run with filter: {filter_query}")
+        runs = mlflow.search_runs(
+            experiment_ids=[experiment_id],
+            filter_string=filter_query,
+            order_by=["start_time"],
+        )
+
+        if len(runs) == 0:
+            print("No run with matching filter found. Creating a new run.")
+            with mlflow.start_run(
+                run_id=self._get_or_create_parent_run_id(monitor_name)
+            ) as current_run:
+                print(f"Current parent run id: {current_run.info.run_id}")
+                run_name = f"{signal_name}_{metric_name}"
+                with mlflow.start_run(
+                    nested=True,
+                    run_name=run_name,
+                    tags=self._create_run_tags(
+                        monitor_name, signal_name, metric_name, groups
+                    ),
+                ) as nested_run:
+                    run_id = nested_run.info.run_id
+
+            print(f"Created child run with id '{run_id}'.")
+        else:
+            run_id = runs.iloc[0].run_id
+            print(f"Found run with id '{run_id}' matching filter.")
+        return run_id
+
+    def publish_metric(self, run_id: str, value: float, threshold, step: int):
+        """Publish a metric to the run metrics store."""
+        metrics = {}
+        metrics["value"] = value
+        if threshold is not None:
+            metrics["threshold"] = float(threshold)
+        self.publish_metrics(run_id=run_id, metrics=metrics, step=step)
+
+    def publish_metrics(self, run_id: str, metrics: dict, step: int):
+        """Publish metrics to the run metrics store."""
+        print(
+            f"Publishing metric {metrics} to run {run_id} at step {step}."
```

Comment:
nit: could you wrap the metrics dictionary like '{metrics}' for readability?

Reply:
May I address this as a follow-up once I'm back? Will change whilst doing the samples work

Reply:
Added this WIT - https://msdata.visualstudio.com/Vienna/_workitems/edit/2659503